client/network-gossip/src/bridge.rs

```diff
@@ -38,7 +38,7 @@ pub struct GossipEngine<B: BlockT> {
 	state_machine: ConsensusGossip<B>,
 	network: Box<dyn Network<B> + Send>,
 	periodic_maintenance_interval: futures_timer::Delay,
-	engine_id: ConsensusEngineId,
+	protocol: Cow<'static, str>,
```

Comment:
An alternative to `Cow` would be something like `smol_str` which is immutable inlined (up to 22 bytes) string with O(1) clone. The downside is the dependency on a third-party lib in a public API.